### ISSUE TITLE
Triple A Bomber Movement

### ISSUE THREAD

**Issue by MarkTheSark** [triplea-game/triplea#4649]
### Can you describe how to trigger the error? (eg: what sequence of actions will recreate it?) American Bombers try to Bomb Japan (or move anywhere) They are currently located in China.
[TripleAZip.zip](https://github.com/triplea-game/triplea/files/2837397/TripleAZip.zip)

**Comment by simon33-2** [comment]
Looks like somehow the movement was not granted at the start of combat movement. I think we'd need an earlier save to be able to proceed with this one. Do you still have the autosave from the end of the Japan turn and if we load it up do we still see the problem?

**Comment by MarkTheSark** [comment]
Thanks for the reply.

In sorry, but all I have is what I sent. I've started new games and resaved
over the same title.

**Comment by simon33-2** [comment]
ok, hopefully someone can figure something.

**Comment by MarkTheSark** [comment]
I really appreciate the help.

**Comment by simon33-2** [comment]
Here's a test case from a little earlier in the cycle which shows the same problem:
[test32707.zip](https://github.com/triplea-game/triplea/files/3033460/test32707.zip)

**Comment by VictoryFirst2** [comment]
@Panther @DanVanAtta Is this still an issue in the latest TripleA version? I have never experienced this error at all, nor seen anyone experience it.

If not, I assume this issue has been fixed somewhere along the way, and I'd suggest to close this.

**Comment by panther2** [comment]
@VictoryFirst1 I have never experienced this either. This issue is 7 years old now - and it looks as if it never has been reproducible - even back in 2019.
I would not mind if we closed it. Let's wait three days, and if no one else will disagree we close it?

**Comment by Cernelius** [comment]
I've installed the latest TripleA (2.7.15614), downloaded and loaded the game-save (I assume that it is the one called "1".) given at the opening post with it. It failed, giving the following information:

>Error loading save game, saved version might not be compatible with current engine.
>
>NullPointerException: Cannot invoke "String.equals(Object)" because the return value of "games.strategy.engine.data.UnitType.getName()" is null

I assume that it is not any longer possible to test this issue with the provided game-save.

**Comment by VictoryFirst2** [comment]
@Cernelius Agreed, it would be better to wait until someone reports this issue again, rather than leaving this open or trying to reproduce it ourselves. Otherwise, we should assume it has been fixed.

**Comment by Cernelius** [comment]
I'll however point out that the information given by TripleA 2.5 is much better than the one given by 2.7.15614 (especially for a regular user). This is what 2.5 gives:

>Incompatible engine versions. We are: 2.5.22294
>Trying to load incompatible save game version: 1.9.0.0.13066
>To download an older version of TripleA,
>please visit: https://triplea-game.org/old_downloads/
>
>If this problem happens frequently and is something you cannot fix,
>please report it to TripleA 

I'm seeing a regression here...
